cgo_ref_pool.go

```diff
@@ -48,15 +48,17 @@ func (refPool *cgoRefPool) AllocCString(str string) uintptr {
 // underlying array in the string value using the nativeStringUnwrap function. Hence, removing any copy in the process
 func (refPool *cgoRefPool) AllocWafString(obj *bindings.WafObject, str string) {
 	obj.Type = bindings.WafStringType
+	emptyStr := len(str) == 0
 
-	if len(str) == 0 {
-		obj.NbEntries = 0
-		obj.Value = 0
-		return
+	if emptyStr {
+		str = "\x00"
 	}
 
 	refPool.stringRefs = append(refPool.stringRefs, str)
 	stringHeader := unsafe.NativeStringUnwrap(str)
+	if emptyStr {
+		stringHeader.Len = 0
+	}
```

Comment:
🤔 Since strings in the WAF are not null-terminated, I am wondering how this is different than before. Sure now `obj.Value` is not null but noone should read any bytes of this pointer because it have a length of zero.

What error is this code fixing exactly?

Reply:
What happens atm is that any empty string `""` gets converted to a null string pointer waf object, but the WAF in some cases only expects empty C strings OR no data at all, but not a null pointer. Such a case happens for the redirect action location parameter, for instance.
This code takes care of passing those empty Go string as empty C strings (so nil terminated).

Reply:
Summarising our offline discussion:
- We don't want to have to pass null terminated strings in some cases since it breaks a big WAF precedent
- We expect the WAF to handle this case properly and will wait for an update to have a clean fix
- Meanwhile, we'll tweak the WAF objects in the case of empty strings so that they don't get tagged as null pointers on the WAF side.